sys/openbsd/extract-openbsd.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/ksh
```

Comment:
Doesn't this work?
```
gmake extract SOURCEDIR=/openbsd/checkout/dir
gmake generate
```
If not, how it does not work?

This script is really not what we want to promote. It does some steps twice, some excessive and unnecessary steps, duplicates too many details and in the end does the same as make extract+generate.

Reply:
It works just fine. Reverted the script. Can we make the default `SOURCEDIR=/usr/src`?

One question, why does `gmake generate` introduce this whitespace diff that Travis then complains about?
```diff
modified   pkg/build/akaros.go
@@ -86,8 +86,8 @@ bash
 		return fmt.Errorf("failed to write image file: %v", err)
 	}
 	for src, dst := range map[string]string{
-		".config": "kernel.config",
-		"key":     "key",
+		".config":                    "kernel.config",
+		"key":                        "key",
 		"obj/kern/akaros-kernel":     "kernel",
 		"obj/kern/akaros-kernel-64b": "obj/akaros-kernel-64b",
 	} {
modified   pkg/vcs/vcs_test.go
@@ -31,13 +31,13 @@ func TestCheckRepoAddress(t *testing.T) {
 		"https://anonscm.debian.org/git/kernel/linux.git":                       true,
 		"git://kernel.ubuntu.com/ubuntu/ubuntu-zesty.git":                       true,
 		"http://host.xz:123/path/to/repo.git/":                                  true,
-		"":           false,
-		"foobar":     false,
-		"linux-next": false,
-		"foo://kernel.ubuntu.com/ubuntu/ubuntu-zesty.git":    false,
-		"git://kernel/ubuntu.git":                            false,
-		"git://kernel.com/ubuntu":                            false,
-		"gitgit://kernel.ubuntu.com/ubuntu/ubuntu-zesty.git": false,
+		"":                                                                      false,
+		"foobar":                                                                false,
+		"linux-next":                                                            false,
+		"foo://kernel.ubuntu.com/ubuntu/ubuntu-zesty.git":                       false,
+		"git://kernel/ubuntu.git":                                               false,
+		"git://kernel.com/ubuntu":                                               false,
+		"gitgit://kernel.ubuntu.com/ubuntu/ubuntu-zesty.git":                    false,
 	})
 }
 
modified   prog/prog_test.go
@@ -223,11 +223,11 @@ func TestEscapingPaths(t *testing.T) {
 		"file/../../file":        true,
 		"../file":                true,
 		"./file/../../file/file": true,
-		"":          false,
-		".":         false,
-		"file":      false,
-		"./file":    false,
-		"./file/..": false,
+		"":                       false,
+		".":                      false,
+		"file":                   false,
+		"./file":                 false,
+		"./file/..":              false,
 	}
 	target, err := GetTarget("test", "64")
 	if err != nil {
```

Reply:
If it's any use, here's the [full log of my running gmake extract SOURCEDIR=/usr/src && gmake generate](https://gist.github.com/blackgnezdo/063b493a3f9f04ee8d75220ab0c3b752)

Reply:
See https://github.com/google/syzkaller/blob/master/docs/contributing.md#how-to-create-a-pull-request re go fmt.

Reply:
> Can we make the default SOURCEDIR=/usr/src

If we do this, at the very least we will also need a good diagnostic for wrong SOURCEDIR in makefile.
But frankly I would prefer to leave it explicit. People don't necessary run the very latest version, e.g. think of cases when we start adding new syscalls as they are merged into kernel source tree, running against /usr/src will be always wrong in that case, and even if you adding an old syscall you now still need the very latest version because all consts are regenerated. Some users don't have sources installed (right?). This also works on linux and they sources are definitely not in /usr/src. And we don't do this for any other OS.